services/motion/plan_status.go

```diff
@@ -0,0 +1,117 @@
+package motion
+
+import (
+	"time"
+
+	pb "go.viam.com/api/service/motion/v1"
+
+	"go.viam.com/rdk/resource"
+)
+
+type PlanWithStatus struct {
```

Comment:
This struct is dubious to me. It contains a Plan which has a `PlanID` as defined below.  Then the PlanStatus type also has a `PlanID`, which can have similar or different PlanIDs in it.  It feels like there should be a simpler way to do this, with less redundancy and that it may be worth rethinking now before we solidify it in the API/RDK